### ISSUE TITLE
Test secret name update for S3 credentials - Test Release 2.20

### ISSUE THREAD

**Issue by ahmedwaleedmalik** [kubermatic/kubermatic#9243]
# Instructions

_TBD_

# References

xref https://github.com/kubermatic/kubermatic/pull/9230

**Comment by ahmedwaleedmalik** [comment]
@lsviben @xrstf can you guys please suggest ways in which this change can/should be tested? I want to make sure we don't miss out on anything.

**Comment by lsviben** [comment]
So you will need to test if the old backups work, as you are changing its secret. Not sure if this affects anything else, maybe metering was using minio as well? (please check)

The old backups are enabled if the new ones are not :D. So you would need to disable/not enable the new backups on  a Seed, meaning that of `seed.spec.etcdbackupRestore` should be nil.

Then this controller should kick in and start making backup cronjobs for each cluster:
https://github.com/kubermatic/kubermatic/blob/release/v2.20/pkg/controller/seed-controller-manager/backup/backup_controller.go

job: 
https://github.com/kubermatic/kubermatic/blob/master/pkg/controller/operator/defaults/defaults.go#L711-L751

which i already see it wont work because its expecting `s3-credentials`.

But in general, if that would work, means that it works.

**Comment by mlavacca** [comment]
@lsviben I'm testing it and setting to nil `seed.spec.etcdbackupRestore` is not enough. Maybe I should enable `seed.spec.backupRestore`? In that case, testing is a bit more difficult, because we don't have such a field in the new CRDs (all the deprecated fields have been removed), and I should test in an environment before the migration.

**Comment by lsviben** [comment]
Ill test on dev why it doesnt work. 

`seed.spec.backupRestore` was also for the new backups, so it wont work. What you need is to trigger the old old backups that needed manual restore, and didnt use EtcdBackupConfig for scheduling automatic backups.

This is where the decision is to enable the old backups:
https://github.com/kubermatic/kubermatic/blob/release/v2.20/pkg/controller/seed-controller-manager/backup/backup_controller.go#L240. 

which checks this:

```// IsDefaultEtcdAutomaticBackupEnabled returns true if etcd automatic backup is configured for the seed.
func (s *Seed) IsDefaultEtcdAutomaticBackupEnabled() bool {
	if cfg := s.Spec.EtcdBackupRestore; cfg != nil {
		return len(cfg.Destinations) > 0 && cfg.DefaultDestination != ""
	}

	return false
}
```
so it should be enough even to have 0 destinations

**Comment by mlavacca** [comment]
Works as expected. I just Corrected the PR #9230 description that marked as required the existence of a secret named `kkp-s3-credentials` instead of `kubermatic-s3-credentials`.
